android/app/src/androidTest/java/com/tourneynizer/tourneynizer/util/CookieRequestFactoryTest.java

```diff
@@ -0,0 +1,111 @@
+package com.tourneynizer.tourneynizer.util;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import com.android.volley.AuthFailureError;
+import com.android.volley.Request;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonArrayRequest;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.StringRequest;
+import com.tourneynizer.tourneynizer.services.HTTPService;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+@RunWith(AndroidJUnit4.class)
+public class CookieRequestFactoryTest {
+
+    private CookieRequestFactory factory;
+
+    @Before
+    public void createFactory() throws Exception {
+        // Context of the app under test.
+        Context context = InstrumentationRegistry.getTargetContext();
+        HTTPService.init(context);
+        factory = new CookieRequestFactory();
+    }
+
+    @Test
+    public void testMakeStringRequest() {
+        StringRequest request = factory.makeStringRequest(Request.Method.GET, "http://www.testurl.com", new Response.Listener<String>() {
+            @Override
+            public void onResponse(String response) {
+
+            }
+        }, new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+
+            }
+        });
+        assertTrue(request != null);
+        Map<String, String> fakeHeader = new HashMap<>();
+        try {
+            assertEquals(fakeHeader, request.getHeaders());
+        } catch (AuthFailureError e) {
+            assertTrue(false);
+        }
+    }
+
+    @Test
+    public void testMakeJsonObjectRequest() {
+        JsonObjectRequest request = factory.makeJsonObjectRequest(Request.Method.GET, "http://www.testurl.com", null, new Response.Listener<JSONObject>() {
+            @Override
+            public void onResponse(JSONObject response) {
+
+            }
+        }, new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+
+            }
+        });
+        assertTrue(request != null);
+        Map<String, String> fakeHeader = new HashMap<>();
+        try {
+            assertEquals(fakeHeader, request.getHeaders());
```

Comment:
Same question. Why not make assert other properties like method and url.

Reply:
Because the url and other things are not part of my factory. My factory only affects the header of the request.